timelock/src/lib.rs

```diff
@@ -87,9 +85,9 @@ impl ClaimableBalanceContract {
             panic!("contract has been already initialized");
         }
 
-        let from_id = from.identifier(&env);
+        verify_and_consume_nonce(&env, &from, &BigInt::zero(&env));
 
-        verify_and_consume_nonce(&env, &from_id, &BigInt::zero(&env));
+        let from_id = from.identifier(&env);
 
         // Authenticate depositor with nonce of zero, so that this may
```

Comment:
This comment is no longer relevant, as 0 nonce doesn't prevent anything for invoker signatures and we have a separate init check anyway.